include/swift/AST/DiagnosticsSema.def

```diff
@@ -1981,16 +1981,16 @@ NOTE(requires_stored_property_inits_here,none,
 ERROR(class_without_init,none,
       "%select{class|actor}0 %1 has no initializers", (bool, Type))
 NOTE(note_no_in_class_init_1,none,
-     "stored property %0 without initial value prevents synthesized "
+     "stored property %0 without initial value prevents default "
```

Comment:
As Jordan notes [here](https://github.com/apple/swift/issues/43376#issuecomment-1108158357), the compiler term *synthesized initializers* encompasses both default *and* inherited initializers.

What if we rearchitect the diagnosis to avoid these terms altogether by pointing out that the class must implement an initializer because it has stored properties without default values? As a bonus, I believe this formulation would make for a far more helpful error message than the current `class has no initializers`. The notes could then be adjusted to simply point at these stored properties instead of spelling out causal relationships.

Reply:
I totally overlooked his comment! Thank you for highlighting it.

I'm not confident in my understanding of the relation with the inherited initializers, so I'll give an example and walk through it to check if I understood correctly.
```Swift
class Superclass {
    let boolean: Bool
    // The compiler will synthesize a default initializer like init(boolean: Bool)
}
class Subclass: Superclass {
    let text: String = "Example Text"
    // The compiler will synthesize an inherited initializer like init(boolean: Bool), like the one seen above
}
```
If the `text` variable didn't have an initial value, the inherited initializer wouldn't be able to be synthesized. But the changes I've made to the diagnostics would call the inherited initializer a default initializer, which is wrong.

Did I use the correct terms everywhere?

The solution then is to check if we are synthesizing an inherited initializer or a default initializer. And your suggestion is to change the current diagnostics to something along the lines of:
`error: an initializer must be implemented because this class has stored properties without initial values`
`note: 'text' has no initial value`

Did I understand the solution you proposed correctly?

Reply:
> If the `text` variable didn't have an initial value, the inherited initializer wouldn't be able to be synthesized. But the changes I've made to the diagnostics would call the inherited initializer a default initializer, which is wrong.

Yes, except a default initializer is an implicit initializer with no parameters. *Memberwise* initializers like `init(boolean: Bool)` are only synthesized only for structures. Inherited initializers can have parameters and are somewhat harder to define in terms of what actually happens. For a designated initializer, the inherited initializer is a distinct, implicit override thereof. Otherwise, it is the superclass initializer itself, and is called inherited because you can call it on the subclass. It’s easiest to simply think of them as initializers that are inherited per initializer inheritance rules.

> Did I understand the solution you proposed correctly?

Sounds good!

Reply:
> `note: 'text' has no initial value`

We spell these sorts of notes as `<description> declared here`. It’s better not to phrase the note as a complaint about the absence of an default value because we don’t know what course of action is best for the programmer, and we’d rather not mislead them.